### PR TITLE
Use async_grpc

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -172,6 +176,7 @@ if(${BUILD_GRPC})
       cartographer/cloud/map_builder_server_main.cc
   )
   target_link_libraries(cartographer_grpc_server PUBLIC grpc++)
+  target_link_libraries(cartographer_grpc_server PUBLIC async_grpc)
```
Comment (gaschler) [inline CMakeLists.txt]: After you did find_package above, you should use the variable ASYNC_GRPC_LIBRARIES here (and below)

File: scripts/install_cartographer_bazel.sh [hunk 2]
```diff
@@ -20,4 +20,4 @@ set -o verbose
 # Build and install Cartographer.
 cd cartographer
 bazel build //...
-bazel test //...
+bazel test --jobs 1 //...
```
Comment (gaschler) [inline scripts/install_cartographer_bazel.sh]: unrelated but ok

File: scripts/install_async_grpc.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o verbose
+
+git clone https://github.com/googlecartographer/async_grpc
+cd async_grpc
+mkdir build
```
Comment (gaschler) [inline scripts/install_async_grpc.sh]: pin a version with git checkout

File: scripts/install_async_grpc.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+cd build
+cmake -G Ninja \
+  -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
```
Comment (gaschler) [inline scripts/install_async_grpc.sh]: is this necessary? bazel doesn't seem to need this

File: cartographer/cloud/internal/local_trajectory_uploader.cc [hunk 5]
```diff
@@ -83,38 +84,38 @@ class LocalTrajectoryUploader : public LocalTrajectoryUploaderInterface {
...
-  std::unique_ptr<framework::Client<handlers::AddLandmarkDataHandler>>
+  std::unique_ptr<async_grpc::Client<handlers::AddLandmarkDataSignature>>
       add_landmark_client_;
 };
 
 LocalTrajectoryUploader::~LocalTrajectoryUploader() {
   if (add_imu_client_) {
-    CHECK(add_imu_client_->WritesDone());
-    CHECK(add_imu_client_->Finish().ok());
+    CHECK(add_imu_client_->StreamWritesDone());
```
Comment (gaschler) [inline cartographer/cloud/internal/local_trajectory_uploader.cc]: unrelated, but fine

File: CMakeLists.txt (hunk 1 above)
Comment (cschuet) [reply CMakeLists.txt]: as discussed.

**Pull request merged**